app/models/case_management/icasework/published_request.rb

```diff
@@ -13,9 +13,62 @@ def reference
         request.case_id
       end
 
+      # TODO: Is there an attribute where we don't need to remove the prefix?
+      def title
+        case_details[:case_label].gsub('IR - ', '')
+      end
+
+      # TODO: We can't really store this since the access token is only valid
+      # for an hour. Can we call out to the library to get the latest URL?
+      #
+      # TODO: We'll also need to be more intellegent about which document to
+      # link to. We'll want to filter by the Name attribute for the one
+      # beginning "Response" (see mysociety/icasework-ruby#10)
+      def url
+        documents.last
+      end
+
+      # TODO: Extract contents of couments
+      def summary
+        case_attributes[:details]
+      end
+
+      # TODO: A classification has a Group attribute and a value. Should we use
+      # all of them?
+      def keywords
+        request[:classifications].join(', ')
+      end
+
+      # TODO: Check whether there's a better attribute available
```

Comment:
We do have a `CaseStageStageN.DateCompleted` , so we could do:

```ruby
request[:case_stage_stage].
  map { |stage| stage[:date_completed] }.compact.
  map { |date| Date.parse(date) }.
  sort.last
```

There's also a `CaseStatusClosure.DateClosed` which seems more like the "publication time", but isn't populated.

Alternatively there's `CaseStatusTouchTimes.LastUpdatedTime`, but that could be updated after publication.

```xml
<Case>
<CaseStatus.CurrentOutcomeDate>2021-02-15</CaseStatus.CurrentOutcomeDate>
<CaseStatus.CurrentStageOutcomeDate>2021-02-15</CaseStatus.CurrentStageOutcomeDate>
<CaseStatusReceipt.DateReceived>2021-02-15</CaseStatusReceipt.DateReceived>
<CaseStatusReceipt.TimeCreated>2021-02-15T16:42:19</CaseStatusReceipt.TimeCreated>
<CaseStatusTouchTimes.FirstTouchTime>2021-02-15T16:42:20</CaseStatusTouchTimes.FirstTouchTime>
<CaseStatusTouchTimes.LastUpdatedTime>2021-03-17T14:20:37</CaseStatusTouchTimes.LastUpdatedTime>
<CaseStatusTouchTimes.LastTouchTime>2021-03-17T14:20:37</CaseStatusTouchTimes.LastTouchTime>
<CaseStatusAcceptance.DateAccepted>2021-02-15</CaseStatusAcceptance.DateAccepted>
<CaseStatusRejection.DateReturned></CaseStatusRejection.DateReturned>
<CaseStatusWithdrawal.DateWithdrawn></CaseStatusWithdrawal.DateWithdrawn>
<CaseStatusClosure.DateClosed></CaseStatusClosure.DateClosed>
<CaseStatusClosure.ClosureInTarget></CaseStatusClosure.ClosureInTarget>
<CaseStageStage1.DateDue>2021-03-15</CaseStageStage1.DateDue>
<CaseStageStage1.DateCompleted>2021-02-15</CaseStageStage1.DateCompleted>
<CaseStageStage2.DateReceived></CaseStageStage2.DateReceived>
<CaseStageStage2.DateDue></CaseStageStage2.DateDue>
<CaseStageStage2.OutcomeBy></CaseStageStage2.OutcomeBy>
<CaseStageStage2.DateCompleted></CaseStageStage2.DateCompleted>
<CaseStageStage3.TargetDays></CaseStageStage3.TargetDays>
<CaseStageStage3.DateReceived></CaseStageStage3.DateReceived>
<CaseStageStage3.DateDue></CaseStageStage3.DateDue>
<CaseStageStage3.Outcome></CaseStageStage3.Outcome>
<CaseStageStage3.OutcomeDetails></CaseStageStage3.OutcomeDetails>
<CaseStageStage3.OutcomeBy></CaseStageStage3.OutcomeBy>
<CaseStageStage3.DateCompleted></CaseStageStage3.DateCompleted>
<CaseStageStage3.ResponseInTarget></CaseStageStage3.ResponseInTarget>
<CaseStageStage4.TargetDays></CaseStageStage4.TargetDays>
<CaseStageStage4.DateReceived></CaseStageStage4.DateReceived>
<CaseStageStage4.DateDue></CaseStageStage4.DateDue>
<CaseStageStage4.Outcome></CaseStageStage4.Outcome>
<CaseStageStage4.OutcomeDetails></CaseStageStage4.OutcomeDetails>
<CaseStageStage4.OutcomeBy></CaseStageStage4.OutcomeBy>
<CaseStageStage4.DateCompleted></CaseStageStage4.DateCompleted>
<CaseStageStage4.ResponseInTarget></CaseStageStage4.ResponseInTarget>
</Case>
```

Reply:
I've [asked about this](https://3.basecamp.com/4020879/buckets/20870999/todolists/3580741982#adding_to_3580741982) on Basecamp.

Reply:
Extracted as new issue https://github.com/mysociety/foi-for-councils/issues/413 so that we don't get caught up on this.